diff -Nur -x '"build"' -x '".cproject"' -x '".project"' -x '".settings"' gcc-7.3.0/config.sub gcc-ncpu32k-7.3.0/config.sub
--- gcc-7.3.0/config.sub	2016-12-29 19:04:53.000000000 +0800
+++ gcc-ncpu32k-7.3.0/config.sub	2019-08-13 16:38:46.716682192 +0800
@@ -295,6 +295,7 @@
 	| moxie \
 	| mt \
 	| msp430 \
+	| ncpu32k \
 	| nds32 | nds32le | nds32be \
 	| nios | nios2 | nios2eb | nios2el \
 	| ns16k | ns32k \
@@ -421,6 +422,7 @@
 	| mmix-* \
 	| mt-* \
 	| msp430-* \
+	| ncpu32k-* \
 	| nds32-* | nds32le-* | nds32be-* \
 	| nios-* | nios2-* | nios2eb-* | nios2el-* \
 	| none-* | np1-* | ns16k-* | ns32k-* \
diff -Nur -x '"build"' -x '".cproject"' -x '".project"' -x '".settings"' gcc-7.3.0/configure gcc-ncpu32k-7.3.0/configure
--- gcc-7.3.0/configure	2017-03-23 01:51:18.000000000 +0800
+++ gcc-ncpu32k-7.3.0/configure	2019-08-13 16:37:52.120680838 +0800
@@ -6873,6 +6873,9 @@
   mips*-*-*linux* | mips*-*-gnu*)
     target_makefile_frag="config/mt-mips-gnu"
     ;;
+  ncpu32k-*-elf*)
+    target_makefile_frag="config/mt-ncpu32k-elf"
+    ;;
   nios2-*-elf*)
     target_makefile_frag="config/mt-nios2-elf"
     ;;
diff -Nur -x '"build"' -x '".cproject"' -x '".project"' -x '".settings"' gcc-7.3.0/configure.ac gcc-ncpu32k-7.3.0/configure.ac
--- gcc-7.3.0/configure.ac	2017-03-23 01:51:18.000000000 +0800
+++ gcc-ncpu32k-7.3.0/configure.ac	2019-08-13 16:37:30.880680311 +0800
@@ -2471,6 +2471,9 @@
   mips*-*-*linux* | mips*-*-gnu*)
     target_makefile_frag="config/mt-mips-gnu"
     ;;
+  ncpu32k-*-elf*)
+    target_makefile_frag="config/mt-ncpu32k-elf"
+    ;;
   nios2-*-elf*)
     target_makefile_frag="config/mt-nios2-elf"
     ;;
diff -Nur -x '"build"' -x '".cproject"' -x '".project"' -x '".settings"' gcc-7.3.0/gcc/config.gcc gcc-ncpu32k-7.3.0/gcc/config.gcc
--- gcc-7.3.0/gcc/config.gcc	2018-01-08 21:39:11.000000000 +0800
+++ gcc-ncpu32k-7.3.0/gcc/config.gcc	2019-08-13 16:44:35.156690831 +0800
@@ -430,6 +430,9 @@
 	extra_objs="frame-header-opt.o"
 	extra_options="${extra_options} g.opt fused-madd.opt mips/mips-tables.opt"
 	;;
+ncpu32k-*-*)
+	cpu_type=ncpu32k
+	;;
 nds32*)
 	cpu_type=nds32
 	extra_headers="nds32_intrinsic.h"
diff -Nur -x '"build"' -x '".cproject"' -x '".project"' -x '".settings"' gcc-7.3.0/gcc/configure gcc-ncpu32k-7.3.0/gcc/configure
--- gcc-7.3.0/gcc/configure	2017-11-21 17:31:12.000000000 +0800
+++ gcc-ncpu32k-7.3.0/gcc/configure	2019-08-13 16:43:41.056689490 +0800
@@ -24058,6 +24058,14 @@
 	tls_first_minor=19
 	tls_as_opt='--fatal-warnings'
 	;;
+  ncpu32k-*-*)
+      conftest_s='
+	.section ".tdata","awT",%progbits
+	nop'
+	tls_first_major=2
+	tls_first_minor=23
+	tls_as_opt='--fatal-warnings'
+	;;
   nios2-*-*)
       conftest_s='
 	.section ".tdata","awT",@progbits'
@@ -27594,7 +27602,7 @@
 # version to the per-target configury.
 case "$cpu_type" in
   aarch64 | alpha | arm | avr | bfin | cris | i386 | m32c | m68k | microblaze \
-  | mips | nios2 | pa | riscv | rs6000 | score | sparc | spu | tilegx \
+  | mips | ncpu32k | nios2 | pa | riscv | rs6000 | score | sparc | spu | tilegx \
   | tilepro | visium | xstormy16 | xtensa)
     insn="nop"
     ;;
diff -Nur -x '"build"' -x '".cproject"' -x '".project"' -x '".settings"' gcc-7.3.0/gcc/configure.ac gcc-ncpu32k-7.3.0/gcc/configure.ac
--- gcc-7.3.0/gcc/configure.ac	2017-11-21 17:31:12.000000000 +0800
+++ gcc-ncpu32k-7.3.0/gcc/configure.ac	2019-08-13 16:42:50.888688246 +0800
@@ -3295,6 +3295,14 @@
 	tls_first_minor=19
 	tls_as_opt='--fatal-warnings'
 	;;
+  ncpu32k-*-*)
+      conftest_s='
+	.section ".tdata","awT",%progbits
+	nop'
+	tls_first_major=2
+	tls_first_minor=23
+	tls_as_opt='--fatal-warnings'
+	;;
   nios2-*-*)
       conftest_s='
 	.section ".tdata","awT",@progbits'
@@ -4772,7 +4780,7 @@
 # version to the per-target configury.
 case "$cpu_type" in
   aarch64 | alpha | arm | avr | bfin | cris | i386 | m32c | m68k | microblaze \
-  | mips | nios2 | pa | riscv | rs6000 | score | sparc | spu | tilegx \
+  | mips | ncpu32k | nios2 | pa | riscv | rs6000 | score | sparc | spu | tilegx \
   | tilepro | visium | xstormy16 | xtensa)
     insn="nop"
     ;;
